documentation/extension.tex

```diff
@@ -1 +1,98 @@
+
+\subsubsection{Macro Detection}
+
+Each file starting with line \emph{MACRO} (arbitrary number of whitespace before and after) is recognized as HLASM.
+
+\subsubsection{Configuration Files Detection}
+
+Every file either defined as a program or as a part of a processor group is recognized as HLASM.
+
+\subsubsection{Wildcards}
+
+Configuration file \emph{pgm\_conf.json} contains a field \emph{alwaysRecognize}, which consists of user-defined wildcards. Every file that satisfies at least one of these wildcards is recognized as HLASM.
+
+\subsubsection{Automatic Language Detection}
+
+Whenever a file is open, its contents are scanned line by line. Whether a file is HLASM or not is determined by a ratio (\# HLASM lines)/(\# all lines).
+
+Recognition is mostly based on a pre-defined set of most used instructions. If a line correctly uses one of these instructions, it is counted as a HLASM line. Continued line of a HLASM line is also a HLASM line.
+Moreover, a HLASM line must not exceed 80 characters.
+
+Comment lines or empty lines are skipped and not counted. 
+
+We tested the detection on ~25.000 HLASM files. The best results were observed using 4/10 ratio, with 88\% true positive recognition and 95\% true negative recognition. 
+
+Because of the indeterminate outcomes, this method is meant to be used as a fall-back in case all previous methods do not suffice.
+
+All detection layers are visualized in \cref{fig08:lang}.
+
+\subsection{Continuation Handling}
+
+Due to historical reasons, HLASM has a 80 character-per-line limitation. Modern languages do not enforce such restriction and therefore IDEs such as VSCode allow the user to extend their lines freely. This causes 2 major inconveniences.
+
+First of all, the user must add the continuation character on a very specific column manually. Secondly, each time the user types in between continuation character and the instruction/parameters, the continuation character is pushed from its requisite position and needs to be moved back, again manually.
+
+To improve this behavior, the extension offers an option to activate \emph{Continuation Handling}. 
+
+The first problem is solved by adding two editor commands \emph{insertContinuation} and \emph{deleteContinuation}, which, when invoked, insert/delete the continuation character on its correct position.
+
+To improve the second problem, the option overrides standard VSCode commands, commonly used when working in editor such as \emph{type}, \emph{deleteLeft}, \emph{deleteRight}, \emph{cut} and \emph{paste}. They offset the continuation character by removing/adding whitespaces in front of it.
+
+
+\subsection{Configuration Prompt}
+
+If the workspace contains a HLASM file, but does not have configurations, the user is prompted to create them. The warning message also offers an option to create templates for them.
+
+\subsection{HLASM Semantic Highlighting}
+
```

Comment:
niekde tu spomenut ze cast z toho kodu nie je nase? aby nas niekto nevykostil za to, lebo takto to vyzera ze cele to rozsirenie je nase aj ked cast nie je

Reply:
netusim ako velmi sa taketo veci hrotia... ale spomenut to asi neublizi